src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php

```diff
@@ -72,15 +95,18 @@ public function addConfiguration(NodeBuilder $node): void
                     ->arrayNode('issuers')
                         ->info('Issuers allowed to generate the token, for validation purpose.')
                         ->isRequired()
-                        ->prototype('scalar')->end()
+                        ->scalarPrototype()->end()
                     ->end()
-                    ->scalarNode('algorithm')
```

Comment:
IMHO we have 2 DX possibilities:
1. keep `algorithm` and `key` options in case of single algorithm and single key, which are automatically converted to `algorithms` and `keyset` options
2. or deprecate them and handle the BC layer with `beforeNormalization`, and always initialize `algorithms` and `keyset` options internally to don't create deprecated objects (cf. OIDCTokenHandler)

WDYT?

Reply:
One question came into my mind after that: if at some point JWEs are supported, we will have other algorithms lists and keyset. What if we name `algorithms` into `verification_algorithms` and `keyset` into `verification_keyset`? (or under a `verification` section)
Anyway, option 2 looks better to me.

Reply:
Like @Spomky, to me it feels natural to look for a JWSignature key and JWEncryption key.
I would deprecate (Vincent's second option) and have  

```yaml
signature:
    algorithms: []
    keyset: []
encryption:
    algorithms: []
    keyset: []
```

Reply:
IMO `signature` seems more a adequat name than `verification`.

However, even if I like this idea, I would prefer to keep it MVP. JWE is not implemented yet. I would suggest to split these configurations in the JWE related Pull Request (once it'll be created).